Libraries/MLXLMCommon/UserInput.swift

```diff
@@ -0,0 +1,137 @@
+// Copyright © 2024 Apple Inc.
+
+import CoreImage
+import Foundation
+import MLX
+
+/// Container for raw user input.
+///
+/// A ``UserInputProcessor`` can convert this to ``LMInput``.
+/// See also ``ModelContext``.
+public struct UserInput: Sendable {
```

Comment:
This is the new representation of a "prompt".  Previously it was a bare `String` or `[Int]` and was left up to the caller to deal with.  Now `UserInput` encapsulates the different types of inputs and lets us add new values (e.g. video frames) over time without changing the underlying API.  A `UserInputProcessor` handles the conversion from `UserInput` into `LMInput` and the `ModelFactory` takes care of supplying the proper implementation.  This is now consistent handling across LLM and VLM.